### ISSUE TITLE
UHCI: Panic due to oneshot_event being triggered twice

### ISSUE THREAD

**Issue by avdgrinten** [managarm/managarm#349]
@borrrden reported the following assertion failure on real hardware:

```
In function raise, file /var/lib/managarm-buildenv/build/system-root/usr/include/async/oneshot-event.hpp:43: Assertion '!raised
```

Full log: [bootlog.txt](https://github.com/managarm/managarm/files/6627077/bootlog.txt)

**Comment by borrrden** [comment]
Almost full log anyway, this is everything I was able to receive after the UART driver started.  Later I will tweak a few things and get the logs from before this event as well.

**Comment by borrrden** [comment]
I wasn't able to successfully retrieve the first part of the log.  Even the other baud rate didn't produce any meaningful words, let alone information.  It looked like ciphertext, without the hallmarks of incorrect baud speed (The sort of "ascii art lines" that happen).  I was able to determine that this happens with or without the keyboard plugged in though.  I have two other USB devices plugged in:

- The USB drive I am using to boot
- Adapter for VGA to HDMI has a USB power source (not sure if that will show up as a device)

Oddly I saw the usb-hid log lines even when I unplugged my keyboard from USB during the limine boot module extract screen.
